pkg/aws/ec2/storage.go

```diff
@@ -0,0 +1,53 @@
+package ec2
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	ec22 "github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/grafana/cloudcost-exporter/pkg/aws/services/ec2"
+)
+
+func ListEBSVolumes(ctx context.Context, client *ec2.EC2) ([]types.Volume, error) {
+	// Paula: do we want to exclude the snapshots? hmm...
+	params := &ec22.DescribeVolumesInput{
+		Filters: []types.Filter{
+			// exclude snapshots
+			{
+				Name:   aws.String("snapshot-id"),
+				Values: []string{""},
+			},
+		},
+	}
+
+	pager := ec22.NewDescribeVolumesPaginator(*client, params)
+	var volumes []types.Volume
+
+	for pager.HasMorePages() {
+		resp, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		volumes = append(volumes, resp.Volumes...)
+		if resp.NextToken == nil || *resp.NextToken == "" {
+			break
+		}
+	}
+
+	return volumes, nil
+}
+
+func NameFromVolume(volume types.Volume) string {
+	pv_name := "kubernetes.io/created-for/pv/name"
+
+	for _, tag := range volume.Tags {
+		if *tag.Key == pv_name {
+			return *tag.Value
+		}
+	}
+	
+	return ""
+}
```

Comment:
This should be a fairly easy method to add tests for. Some scenarios
- volume being nil
- volume set but no matching tag
- volume set with a matching tag

Reply:
New commit has those in place :sisi: